app/domain/authentication/security/extract_request_origin_ip_addr_info.rb

```diff
@@ -0,0 +1,54 @@
+require 'command_class'
+
+module Authentication
+  module Security
+    ExtractRequestOriginIpAddrInfo = CommandClass.new(
+      dependencies: {
+        # InputValidation
+      },
+      inputs: [:request]
+    ) do
+
+      extend Forwardable
+      def_delegators :@request, :remote_addr, :headers
+
+      def call
+        extract_request_origin_addr_info
+      end
+
+      private
+
+      X_FORWARDED_FOR_HEADER_NAME = 'X-Forwarded-For'
+
+      def extract_request_origin_addr_info
+        xff_header = headers[X_FORWARDED_FOR_HEADER_NAME]
+        xff_header_ip_list = (nil_or_empty? xff_header) ? [] : xff_header.to_s.split(/, /).uniq
```

Comment:
@eranha probably the whole feature will be deleted, but still i think there is a possible bug here,
If you first perform `.uniq` and then extract the first ip from the list, there is a possible bug where the first ip will be deleted before the extraction